server/authorization.go

```diff
@@ -214,47 +259,109 @@ func (a *AuthorizationServer) Check(ctx gocontext.Context, req *authv3.CheckRequ
 	authContext, err := a.handler.authMan.Authenticate(rootContext, apiKey, claims, a.handler.apiKeyClaim)
 	switch err {
 	case auth.ErrNoAuth:
-		return a.unauthenticated(req, envRequest, tracker, api), nil
+		return a.handleFault(
+			req,
+			envRequest,
+			tracker,
+			api,
+			/* authContext= */ nil,
+			fault.NewAdapterFaultWithRpcCode(rpc.UNAUTHENTICATED)), nil
 	case auth.ErrBadAuth:
-		return a.denied(req, envRequest, tracker, authContext, api), nil
+		return a.handleFault(
+			req,
+			envRequest,
+			tracker,
+			api,
+			authContext,
+			fault.NewAdapterFaultWithRpcCode(rpc.PERMISSION_DENIED)), nil
 	case auth.ErrInternalError:
-		return a.internalError(req, envRequest, tracker, err), nil
+		log.Errorf("encountered internal error: %v", err)
+		return a.handleFault(
+			req,
+			envRequest,
+			tracker,
+			/* api= */ "",
+			/* authContext= */ nil,
+			fault.NewAdapterFaultWithRpcCode(rpc.INTERNAL)), nil
 	case auth.ErrNetworkError:
 		if envRequest != nil && envRequest.GetConsumerAuthorization().FailOpen {
 			log.Debugf("FailOpen on operation: %v", envRequest.GetOperation().Name)
 			if err := envRequest.PrepareVariables(); err != nil {
 				log.Errorf("failed to populate context variable: %v", err)
-				return a.denied(req, envRequest, tracker, authContext, api), nil
+				return a.handleFault(
+					req,
+					envRequest,
+					tracker,
+					api,
+					authContext,
+					fault.NewAdapterFaultWithRpcCode(rpc.PERMISSION_DENIED)), nil
 			}
 			return a.authOK(req, tracker, authContext, api, envRequest), nil
 		} else {
-			return a.internalError(req, envRequest, tracker, err), nil
+			return a.handleFault(
+				req,
+				envRequest,
+				tracker,
+				/* api= */ "",
+				/* authContext= */ nil,
+				fault.NewAdapterFaultWithRpcCode(rpc.INTERNAL)), nil
```

Comment:
Codecov indicates these lines are not covered by a test. Can you add a test, please?

Reply:
Done.